Sources/Sentry/SentryTracer.m

```diff
@@ -287,7 +302,19 @@ - (void)captureTransaction
         }
     }];
 
-    [_hub captureTransaction:[self toTransaction] withScope:_hub.scope];
+    SentryTransaction *transaction = [self toTransaction];
+    NSMutableArray<SentryEnvelopeItem *> *additionalEnvelopeItems = [NSMutableArray array];
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+    if (_profiler != nil) {
+        SentryEnvelopeItem *profile = [_profiler buildEnvelopeItemForTransaction:transaction];
+        if (profile != nil) {
+            [additionalEnvelopeItems addObject:profile];
+        }
+    }
```

Comment:
`l`: That's a pragmatic approach of merging the transaction with the profiling data. We usually pass objects down to the client or transport and then serialize them there. This will serialize all your data on the calling thread finishing the transaction. At some point, we should move the serialization to a background thread in the transport, but you don't need to fix this now. I just wanted to point this out, so more people are aware. I think it's fine now.

Reply:
@philipphofmann I have it on my plate to re-work the serialization here anyway because as @bruno-garcia just the data structures themselves are inefficient, I will rework this as part of that follow up task.